tests/Benchmark/EntryPoint.cpp

```diff
@@ -0,0 +1,152 @@
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <windows.h>
+#include <TestFramework.h>
+#include "BenchmarkTestPublisher.h"
+#include <fstream>
+#include <algorithm>
+
+#ifdef WIN32
+#include <wrl\wrappers\corewrappers.h>
+using namespace Microsoft::WRL::Wrappers;
+#endif
+
+using namespace benchmark;
+struct __Result {
+    std::string testName;
+    Microseconds totalRuntime;
+    Microseconds meanRuntime;
+    Microseconds stdDeviation;
+    size_t runCount;
+};
+
+class TestPublisherBase : public BenchmarkTestPublisher {
+public:
+    // Expect results.size() > 1
+    virtual void RegisterTestResults(const std::string& testName, const std::vector<Microseconds>& results) {
+        // Algorithm for calculating sample mean and variance with minimal rounding errors
+        Microseconds netRuntime = 0.0;
+        Microseconds mean = 0.0;
+        Microseconds q = 0.0;
+        size_t runCount = results.size();
+        for (size_t i = 0; i < runCount; ++i) {
+            Microseconds previousMean = mean;
+            netRuntime += results[i];
+            mean += (results[i] - mean) / (i + 1);
+            q += (results[i] - previousMean) * (results[i] - mean);
+        }
+        Microseconds sampleStandardDeviation = std::sqrt(q / (runCount - 1));
+        m_results.emplace_back(__Result{ testName, netRuntime, mean, sampleStandardDeviation, runCount });
+    }
+
+protected:
+    std::vector<__Result> m_results;
+};
+
+class CSVBenchmarkTestPublisher : public TestPublisherBase {
+    std::string m_outPath;
+
+public:
+    CSVBenchmarkTestPublisher(std::string&& outPath) : m_outPath(outPath) {
+    }
+
+    virtual void PublishTestResults() {
+        std::ofstream outFile(m_outPath);
+        outFile << "Test Name"
+                << ","
+                << "Total Runtime"
+                << ","
+                << "Mean Runtime"
+                << ","
+                << "Standard Deviation"
+                << ","
+                << "Run Count"
+                << "\n";
+        for (auto res : m_results) {
+            outFile << res.testName << "," << res.totalRuntime << "," << res.meanRuntime << "," << res.stdDeviation << "," << res.runCount
+                    << "\n";
+        }
+
+        outFile.flush();
+        outFile.close();
+    }
+};
+
+class LogBenchmarkTestPublisher : public TestPublisherBase {
+    size_t m_maxNameWidth = 0;
+
+public:
+    virtual void RegisterTestResults(const std::string& testName, const std::vector<Microseconds>& results) {
+        m_maxNameWidth = max(m_maxNameWidth, testName.size());
+        TestPublisherBase::RegisterTestResults(testName, results);
+    }
+
+    virtual void PublishTestResults() {
+        // Align output horizontally
+        std::string spaces(m_maxNameWidth - 5, ' ');
+        std::cout << "Test Name" << spaces << "|"
+                  << "Total Runtime"
```

Comment:
`std::setw`!

---
In reply to: [99246724](https://github.com/Microsoft/WinObjC/pull/1876#discussion_r99246724) [](ancestors = 99246724)